Task2.2/Program.cs

```diff
@@ -0,0 +1,110 @@
+﻿public class DiagonalMatrix
+{
+    private int[] diagonalElements;
+
+    public int Size { get; }
+
+    public DiagonalMatrix(params int[] elements)
+    {
+        if (elements == null)
+        {
+            Size = 0;
+            diagonalElements = new int[0];
+        }
+        else
+        {
+            Size = elements.Length;
+            diagonalElements = new int[Size];
+            Array.Copy(elements, diagonalElements, Size);
+        }
+    }
+
+    public int this[int i, int j]
+    {
+        get
+        {
+            if (i < 0 || i >= Size || j < 0 || j >= Size || i != j)
+            {
+                return 0;
+            }
+            return diagonalElements[i];
+        }
+        set
+        {
+            if (i >= 0 && i < Size && j >= 0 && j < Size && i == j)
+            {
+                diagonalElements[i] = value;
+            }
+        }
+    }
+
+    public int Track()
+    {
+        int sum = 0;
+        for (int i = 0; i < Size; i++)
+        {
+            sum += diagonalElements[i];
+        }
+        return sum;
+    }
+
+    public override bool Equals(object obj)
+    {
+        if (obj == null || GetType() != obj.GetType())
+            return false;
+

```

Comment:
DiagonalMatrix other = obj as DiagonalMatrix (better). Implementations of [Equals] must not throw exceptions; they should always return a value.